homeassistant/components/sleepiq/button.py

```diff
@@ -0,0 +1,56 @@
+"""Support for SleepIQ buttons."""
+from __future__ import annotations
+
+from asyncsleepiq import SleepIQBed
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SleepIQDataUpdateCoordinator
+from .entity import SleepIQEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sleep number buttons."""
+    coordinator: SleepIQDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[SleepIQEntity] = []
+    for bed in coordinator.client.beds.values():
+        entities.append(SleepNumberCalibrateButton(bed))
+        entities.append(SleepNumberStopPumpButton(bed))
```

Comment:
Instead of creating a class for each button, you could leverage `ButtonEntityDescrition`s instead.

Reply:
Example: https://github.com/home-assistant/core/blob/dev/homeassistant/components/bond/button.py